### PR TITLE
[TBL] Call collex directly for reports

### PR DESCRIPTION
**Pull request by JamesGardiner**
# Motivation and Context

Removes the call to backstage for getting chasing reports.

# What has changed

Changes the `download_report` method in collection_exercise_controller.py to hit the collection exercise endpoint directly.

# How to test?

Run the unit tests, make sure they pass.
Run the unit tests, make sure they pass.
Tail the logs for backstage, collection-exercise service. Hit the `response_chasing/<ce_id>/<survey_id>` endpoint using `curl` or Postman. Make sure backstage doesn't get the call, but collection-exercise service does.

### CODE REVIEW

File: response_operations_ui/controllers/collection_exercise_controllers.py [hunk 1]
```diff
@@ -11,198 +11,311 @@
...
 
 
 def get_case_group_status_by_collection_exercise(case_groups, collection_exercise_id):
-    return next(case_group['caseGroupStatus'] for case_group in case_groups
-                if case_group['collectionExerciseId'] == collection_exercise_id)
+    return next(
+        case_group["caseGroupStatus"]
+        for case_group in case_groups
+        if case_group["collectionExerciseId"] == collection_exercise_id
+    )
```
Comment (jcox-dev) [inline response_operations_ui/controllers/collection_exercise_controllers.py]: 👍

File: config.py [hunk 2]
```diff
@@ -44,6 +44,11 @@ class Config(object):
     PARTY_PASSWORD = os.getenv('PARTY_PASSWORD')
     PARTY_AUTH = (PARTY_USERNAME, PARTY_PASSWORD)
 
+    RM_REPORT_SERVICE_HOST = os.getenv('RM_REPORT_SERVICE_HOST', 'localhost')
+    RM_REPORT_SERVICE_PORT = os.getenv('RM_REPORT_SERVICE_PORT', 8084)
+    RM_REPORT_SERVICE_PROTOCOL = os.getenv('RM_REPORT_SERVICE_PROTOCOL', 'http')
+    RM_REPORT_SERVICE = f'{RM_REPORT_SERVICE_PROTOCOL}://{RM_REPORT_SERVICE_HOST}:{RM_REPORT_SERVICE_PORT}/'
```
Comment (jcox-dev) [inline config.py]: Should be using the new NAME_URL service form with defaults (only?) in the DevelopmentConfig.

Do we also need separate auth details for the report service?

Comment (benjefferies) [reply config.py]: I presume to see the env vars to be condensed

**Comment by JamesGardiner** [comment]
@benjefferies and @jcox-dev - I've addressed your comments. There is no concept of basic auth yet for report service. This should be added by @AdamHawtin and co when it is implemented.

File: config.py [hunk 3]
```diff
@@ -44,6 +44,11 @@ class Config(object):
     PARTY_PASSWORD = os.getenv('PARTY_PASSWORD')
     PARTY_AUTH = (PARTY_USERNAME, PARTY_PASSWORD)
 
+    REPORT_HOST = os.getenv('REPORT_HOST')
+    REPORT_PORT = os.getenv('REPORT_PORT')
+    REPORT_PROTOCOL = os.getenv('REPORT_PROTOCOL')
+    REPORT_URL = f'{REPORT_PROTOCOL}://{REPORT_HOST}:{REPORT_PORT}/'
```
Comment (benjefferies) [inline config.py]: @JamesGardiner I was thinking a bit more along the lines of how it's done for SECURE_MESSAGE_URL

**Comment by jcox-dev** [comment]
[response-operations-ui should directly call other services (not via backstage)](https://trello.com/c/0lPdzRbt/144-response-operations-ui-should-directly-call-other-services-not-via-backstage)

File: config.py [hunk 4]
```diff
@@ -44,6 +44,11 @@ class Config(object):
     PARTY_PASSWORD = os.getenv('PARTY_PASSWORD')
     PARTY_AUTH = (PARTY_USERNAME, PARTY_PASSWORD)
 
+    REPORT_HOST = os.getenv('REPORT_HOST')
```
Comment (jcox-dev) [inline config.py]: @JamesGardiner still need to update the standard Config to something like:

```python
REPORT_URL = os.getenv('REPORT_URL')
```

**Comment by JamesGardiner** [comment]
Cheers @benjefferies

**Pull request merged**